### PR TITLE
output mode && notebook handle in bokeh

### PR DESCRIPTION
**Pull request by Bartolo1024**

### CODE REVIEW

**Comment by stared** [comment]
Can I merge, or should I wait for more updates?

**Comment by Bartolo1024** [comment]
> Can I merge, or should I wait for more updates?

Now you can merge it, if I should not spend too much time on it. I have just fix the script mode. Currently, notebook mode has real time plotting. Unfortunately in browser it requires refresh.

**Comment by stared** [comment]
I am not sure if I understand that.

A short question is: does it work in Colab?

**Comment by Bartolo1024** [comment]
Yes

File: tests/test_plot_losses.py [hunk 1]
```diff
@@ -1,9 +1,8 @@
...
-
 from livelossplot import MainLogger, PlotLosses
+from livelossplot.outputs import BaseOutput
 
 
-class CheckOutput:
+class CheckOutput(BaseOutput):
     def send(self, logger: MainLogger):
         assert isinstance(logger, MainLogger)
         grouped_log_history = logger.grouped_log_history(raw_names=True, raw_group_names=True)
```
Comment (stared) [inline tests/test_plot_losses.py]: I don't understand this part (on any level, including why `CheckOutput` is a kind of output).

**Comment by Bartolo1024** [comment]
Ok, so now should be fine. I've changed tmp to workdir. In the case of root - livelossplot it will fail because of access rights.

File: livelossplot/plot_losses.py [hunk 2]
```diff
@@ -11,13 +11,17 @@ class PlotLosses:
     """
     Class collect metrics from the training engine and send it to plugins, when send is called
     """
-    def __init__(self, outputs: Optional[List[Type[BO]]] = None, **kwargs):
+    def __init__(self, outputs: Optional[List[Type[BO]]] = None, mode: str = 'notebook', **kwargs):
         """
         :param outputs: list of callbacks (outputs) which are called with send method
+        :param mode: mode for callbacks - some of outputs need to change some behaviors,
+         depending on the working environment (scripts and jupyter notebooks)
```
Comment (stared) [inline livelossplot/plot_losses.py]: Still, it needs to be explicit. `Options: 'notebook' or 'script'` or something in that line.

File: tmp.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            'val_acc': 1 - np.random.rand() / (i + 0.5),
+            'loss': 1. / (i + 2.),
+            'val_loss': 1. / (i + 0.5),
+            'mse': 1. / (i + 0.8),
+            'val_mse': 1. / (i + 0.7),
+            'lr': 0.01
+        }
+    )
+    liveplot.send()
+    sleep(.1)
```
Comment (stared) [inline tmp.py]: Is this file supposed to be there?

File: tests/test_bokeh_output.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import numpy as np
+
+from livelossplot import PlotLosses
+from livelossplot.outputs import BokehPlot
+
+
+def test_tensorboard():
+    logger = BokehPlot()
+
+    liveplot = PlotLosses(outputs=(logger, ), mode='script')
```
Comment (stared) [inline tests/test_bokeh_output.py]: `[logger]`, not `(logger, )` vide the tuple vs list discussion (and types).

File: tmp.py (hunk 3 above)
Comment (Bartolo1024) [reply tmp.py]: no

File: livelossplot/plot_losses.py (hunk 2 above)
Comment (Bartolo1024) [reply livelossplot/plot_losses.py]: oki

**Pull request merged**